tensorrt_llm/_torch/auto_deploy/utils/pattern_matcher_utils.py

```diff
@@ -0,0 +1,229 @@
+def trace_to_gm(fn: Callable, args: Sequence[torch.Tensor]) -> GraphModule:
+    """
+    Exports a function or Module into a GraphModule via torch_export_to_gm.
+    """
+    module = fn if isinstance(fn, torch.nn.Module) else WrapperModule(fn)
+    return torch_export_to_gm(module, tuple(args))
+
+
+def _not_implemented(*args: Any, **kwargs: Any) -> NoReturn:
+    raise NotImplementedError
+
+
+def _return_true(match: Match) -> bool:
+    return True
+
+
+def register_pattern(
+    search_fn: Callable,
+    replace_fn: Callable,
+    patterns: PatternMatcherPass,
+    dummy_args: List[torch.Tensor],
+    trace_fn: Callable[[Callable, Sequence[torch.Tensor]], GraphModule] = trace_to_gm,
+    example_inputs: Iterable[Any] = [None],
+    exclusive_arg_names: Sequence[str] = (),
+    scalar_workaround: Optional[Any] = None,
+    op_ignore_types: Optional[Mapping[Callable[..., Any], Sequence[type[Any]]]] = None,
+    extra_check: Callable[[Match], bool] = _return_true,
+) -> None:
+    """
+    Tracing a Python-level pattern into a GraphModule and registering its replacement.
+
+    The inductor matcher treats tensor-shaped arguments and numeric (literal) arguments differently:
+
+    Note:
+    1. Numeric (integer/float) args will be lifted as hard-coded literals in the FX graph,
+        utilize `scalar_workaround` to detect those literals and replace with args in the pattern
+    2. register_replacement can auto-generate `search_fn_pattern` if you omit it,
+        but that approach will fail when symbolic shapes are involved. Here
+        we explicitly trace & convert via `fx_to_pattern`.
+
+    Args:
+        search_fn:          Function defining the “before” pattern (traced for matching).
+        replace_fn:         Function or op defining the “after” replacement.
+        patterns:           PatternMatcherPass instance (passed to register_replacement).
+        example_inputs:     Example inputs for register_replacement's initial trace.
+        dummy_args:         Inputs matching search_fn's signature used to generate
+                            the FX pattern (must reflect any real dims materialized
+                            as aten-op literals).
+        trace_fn:           Callable to export Python functions/modules into FX (e.g.
+                            torch_export_to_gm).
+        exclusive_arg_names:
+                            Parameter names to ignore when matching.
+        scalar_workaround:  Optional dict or value to workaround FX scalar lifting bugs.
+        op_ignore_types:    Ignore certain types of arg for certain op type
+    """
+    argnames = list(inspect.signature(search_fn).parameters.keys())
+    specific_gm = trace_fn(search_fn, dummy_args)
+    pattern = fx_to_pattern_with_op_ignore(
+        specific_gm,
+        argnames=argnames,
+        op_ignore_types=op_ignore_types,
+        exclusive_arg_names=exclusive_arg_names,
+        scalar_workaround=scalar_workaround,
+    )
+
+    register_replacement(
```

Comment:
Can we expose the remaining arguments of `register_replacement` so they can be modified from `register_pattern`. E.g., `skip_duplicates`